test/integration/tags/if_else_tag_test.rb

```diff
@@ -35,6 +35,26 @@ def test_if_or
     assert_template_result('',      '{% if a or b or c %} YES {% endif %}', 'a' => false, 'b' => false, 'c' => false)
   end
 
+  def test_if_or_not
+    assert_template_result(' YES ', '{% if a or not b %} YES {% endif %}', 'a' => true, 'b' => true)
+    assert_template_result(' YES ', '{% if a or not b %} YES {% endif %}', 'a' => true, 'b' => false)
+    assert_template_result('',      '{% if a or not b %} YES {% endif %}', 'a' => false, 'b' => true)
+    assert_template_result(' YES ', '{% if a or not b %} YES {% endif %}', 'a' => false, 'b' => false)
+
+    assert_template_result(' YES ', '{% if a or not b or not c %} YES {% endif %}', 'a' => false, 'b' => false, 'c' => true)
+    assert_template_result(' YES ', '{% if a or not b or not c %} YES {% endif %}', 'a' => false, 'b' => false, 'c' => false)
+  end
+
+  def test_if_and_not
+    assert_template_result('',      '{% if a and not b %} YES {% endif %}', 'a' => true, 'b' => true)
+    assert_template_result(' YES ', '{% if a and not b %} YES {% endif %}', 'a' => true, 'b' => false)
+    assert_template_result('',      '{% if a and not b %} YES {% endif %}', 'a' => false, 'b' => true)
+    assert_template_result('',      '{% if a and not b %} YES {% endif %}', 'a' => false, 'b' => false)
+
+    assert_template_result(' YES ', '{% if a and not b and not c %} YES {% endif %}', 'a' => true, 'b' => false, 'c' => false)
+    assert_template_result('',      '{% if a and not b and not c %} YES {% endif %}', 'a' => false, 'b' => false, 'c' => false)
+  end
+
```

Comment:
I think you should test if it supports `{% if true and not 'a' != 'a' %}` which should be true.
Most of the time the conditions are not simple variables, but complex dot-lookups and comparisons.